eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/CollectionResultDefinition.cs

```diff
@@ -4,46 +4,125 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
 using Microsoft.TypeSpec.Generator.ClientModel.Providers;
+using Microsoft.TypeSpec.Generator.Expressions;
 using Microsoft.TypeSpec.Generator.Input;
 using Microsoft.TypeSpec.Generator.Primitives;
 using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Snippets;
+using Microsoft.TypeSpec.Generator.Statements;
 using static Microsoft.TypeSpec.Generator.Snippets.Snippet;
 
 namespace Azure.Generator.Providers
 {
     internal class CollectionResultDefinition : TypeProvider
     {
         private readonly ClientProvider _client;
+        private readonly FieldProvider _clientField;
         private readonly InputOperation _operation;
         private readonly CSharpType _itemModelType;
+        private readonly CSharpType _responseType;
         private readonly bool _isAsync;
+        private readonly InputPagingServiceMetadata _paging;
+        private readonly IReadOnlyList<FieldProvider> _requestFields;
+        private readonly string _scopeName;
+        private readonly string _itemsPropertyName;
+        private readonly string? _nextLinkPropertyName;
+        private readonly InputResponseLocation? _nextPageLocation;
+        private readonly string _getNextResponseMethodName;
 
-        private static readonly ParameterProvider ContinuationTokenParameter =
-            new("continuationToken", $"The continuation token.", new CSharpType(typeof(string)))
-            {
-                DefaultValue = Null
-            };
+        private readonly IReadOnlyList<ParameterProvider> _createRequestParameters;
+        private readonly FieldProvider? _contextField;
+
+        private static readonly ParameterProvider NextLinkParameter =
+            new("continuationToken", $"The continuation token.", new CSharpType(typeof(string), isNullable: true));
```

Comment:
If this is for next link scenarios, why are we calling it continuationToken?

Reply:
actually it should be continuation token, because of Azure.Core contract
https://github.com/Azure/azure-sdk-for-net/blob/da77411961d66ebfb233146e08051ba463a2a9b8/sdk/core/Azure.Core/src/Pageable.cs#L44-L61

Reply:
So, this is not only for next-link scenario, Azure.Core.Pageable.AsPages will handle both next-link and continuation-token.
I will update to nextPage as parameter name for both.